tools/CouchDB_Confirm_Integrity.php

```diff
@@ -65,55 +65,64 @@ function run()
             "sessions",
             array("reduce" => "false")
         );
-        print "Sessions:";
+        print "Sessions:\n";
         $activeExists = $this->SQLDB->prepare(
-            "SELECT count(*) FROM candidate".
-            " c LEFT JOIN session s USING (CandID) WHERE s.Active='Y' AND".
-            " c.Active='Y' AND c.PSCID=:PID and s.Visit_label=:VL"
+            "SELECT count(*) AS count FROM 
+        candidate c LEFT JOIN session s USING (CandID) WHERE s.Active='Y' 
+        AND c.Active='Y' AND c.PSCID=:PID and s.Visit_label=:VL"
         );
         foreach ($sessions as $row) {
             $pscid = $row['key'][0];
             $vl    = $row['key'][1];
             $sqlDB = $this->SQLDB->pselectRow(
-                "SELECT c.*, s.Visit_label, s.Active
+                "SELECT c.*, s.Visit_label, s.Active, c.Active as cActive
                 FROM candidate c
                 LEFT JOIN session s USING (CandID)
                 WHERE c.PSCID=:PID AND s.Visit_label=:VL",
                 array(
-                 "PID" => $pscid,
-                 "VL"  => $vl,
+                    "PID" => $pscid,
+                    "VL" => $vl
                 )
             );
 
-            if ($sqlDB['Active'] != 'Y') {
+            if (!empty($sqlDB) && $sqlDB['cActive'] == 'N') {
+                print "PSCID $pscid is inactive but $row[id] still exists. 
+                Deleting Doc.\n";
+
+                $this->CouchDB->deleteDoc($row['id']);
+            }
+
+            if (!empty($sqlDB) && $sqlDB['Active'] != 'Y') {
```

Comment:
Is that going to affect the functionality or just make it cleaner?

(I'm not sure what happens in our code if you try and delete an already deleted CouchDB doc.)